packages/cryptoplease_link/jupiter_aggregator/lib/src/client.dart

```diff
@@ -24,7 +26,30 @@ abstract class JupiterAggregatorClient {
 
   /// Get swap serialized transactions for a route
   @POST('/swap')
-  Future<JupiterSwapTransactions> getSwapTransactions(
-    @Body() SwapRequestDto swapRequestDto,
+  Future<JupiterSwapResponseDto> getSwapTransactions(
+    @Body() JupiterSwapRequestDto swapRequestDto,
+  );
+}
+
+/// For docs head to https://docs.jup.ag/jupiter-api/price-api-for-solana
+@RestApi(baseUrl: 'https://price.jup.ag/v1')
+abstract class JupiterPriceClient {
+  factory JupiterPriceClient() => _JupiterPriceClient(
+        Dio()
+          ..interceptors.add(
+            InterceptorsWrapper(
```

Comment:
Why do you need this interceptor?

Reply:
Jupiter Price has not a content-type header in the response, so `Dio` cannot infer that the payload is a Json, resulting in a type error when calling `PriceResponseDto.fromJson`. Seems like `retrofit` team won't address this issue and that's the solution I found out:  https://github.com/trevorwang/retrofit.dart/issues/276